grails-app/utils/org/broadinstitute/orsp/utils/DataUseRestrictionParser.groovy

```diff
@@ -56,6 +56,22 @@ class DataUseRestrictionParser {
         dataUseRestriction.cloudStorage = params.cloudStorage
         dataUseRestriction.irb = IssueUtils.getBooleanForParam(params.irb)
         dataUseRestriction.geographicalRestrictions = params.geographicalRestrictions
+
+        dataUseRestriction.noRestriction = IssueUtils.getBooleanForParam(params.noRestriction)
+
+        if (dataUseRestriction.noRestriction) {
+            dataUseRestriction.generalUse = true
+        }
+
+        dataUseRestriction.collaborationInvestigators = IssueUtils.getBooleanForParam(params.collaborationInvestigators)
+        dataUseRestriction.publicationResults = IssueUtils.getBooleanForParam(params.publicationResults)
+        dataUseRestriction.genomicResults = IssueUtils.getBooleanForParam(params.genomicResults)
+        if (dataUseRestriction.genomicResults) {
+            dataUseRestriction.genomicSummaryResults = (String)params.genomicSummaryResults
```

Comment:
nit: casting. `params` is a [GrailsParameterMap](http://docs.grails.org/3.1.3/api/grails/web/servlet/mvc/GrailsParameterMap.html) which inherits from [AbstractTypeConvertingMap](http://docs.grails.org/3.1.3/api/grails/util/AbstractTypeConvertingMap.html) and therefore has some built in casting methods that should help clean up this kind of thing.

Reply:
Aa a side note, I have come to distrust a lot of the dynamic accessors of map keys like this (i.e. `params.genomicResults`) and prefer to use the getters to get at the values and check for null if necessary. Using the null-safe operator (`?`) is helpful for sitations like this too.

Reply:
I quite don't get this. `IssueUtils` returns `true` or `false` if get "Yes" - "No". If I use, for instance, `params.get("generalUse")` I'll get "Yes" or "No", but it's needed a boolean value.

Reply:
Let me test some code ... I'll give you a better example. I know you need a string, but the cast is the code-smell here. I'm pretty sure there's a cleaner way to do this without a hard cast.

Reply:
Ahhh ... this is the real problem:
```
static DataUseRestriction fromParams(DataUseRestriction dataUseRestriction, Object params)
```
I thought this was a grails parameter map, but it's not, it's a plain `Object`. In DataUseController.save() ... you have a `GrailsParameterMap` that is passed in as the object. In `fromParams`, you change that from `GrailsParameterMap` to `Object`. Because you do that, you're losing a ton of built in functionality that Grails does for you for free. You end up re-implementing functions that have already been implemented for you when you up-cast the map to an object.

Reply:
So, looking more closely at this class - there are two ways that this is used. One is that it gets a `GrailsParameterMap` and the other is that it gets a JSON object. I do think the two should be distinctly different, but that's a good amount of re-work that is outside of the scope of this PR. I suggest a tech debt story to clean up usages of this class.

Reply:
OK, agree. I'll create the story as a tech debt then.

Reply:
Here https://broadinstitute.atlassian.net/browse/BTRX-697